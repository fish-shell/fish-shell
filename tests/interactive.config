# vim: set filetype=fish sw=4 ts=4 et:

# source the non-interactive config
source $XDG_CONFIG_HOME/../tmp.config/fish/config.fish

set -g prompt_counter 0
set -g prompt_counter_incr 0
function fish_prompt
    if test $prompt_counter_incr -eq 1
        set -g prompt_counter (expr $prompt_counter + 1)
        set -g prompt_counter_incr 0
    end
    echo "prompt $prompt_counter>"
end
function fish_prompt_event --on-event fish_prompt
    set -g prompt_counter_incr 1
end

set -g fish_greeting ''

function fish_title
end

function _quote
    echo \'(echo $argv[1] | sed -e 's/\'/\\\\\'/g')\'
end

function _echo_var --no-scope-shadowing -d '_echo_var varname [guardval]'
    if set -q argv[2]; echo "@GUARD:$argv[2]@"; end
    set -l var $argv[1]
    switch (count $$var)
    case 0
        echo "\$$var has no value"
    case 1
        set -l IFS ''
        echo "\$$var:" (_quote $$var)
    case \*
        echo "\$$var:"
        for i in (seq (count $$var))
            set -l IFS ''
            echo "$i:" (_quote $$var[1][$i])
        end
    end
    if set -q argv[2]; echo "@/GUARD:$argv[2]@"; end
end

function _marker -d '_marker string - prints @MARKER:$string@'
    echo "@MARKER:$argv[1]@"
end
