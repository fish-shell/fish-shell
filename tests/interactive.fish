#!/usr/local/bin/fish
#
# Interactive tests using `expect`

function die
    echo $argv[1] >&2
    exit 1
end

for i in *.expect
    rm -Rf tmp.interactive.config; or die "Couldn't remove tmp.interactive.config"
    mkdir -p tmp.interactive.config/fish; or die "Couldn't create tmp.interactive.config/fish"
    cp interactive.config tmp.interactive.config/fish/config.fish; or die "Couldn't create tmp.interactive.config/fish/config.fish"

    begin
        set -lx XDG_CONFIG_HOME $PWD/tmp.interactive.config
        set -lx TERM dumb
        expect -n -c 'source interactive.expect.rc' -f $i >tmp.out ^tmp.err
    end
    set -l tmp_status $status
    set res ok
    if not diff tmp.out $i.out >/dev/null
        set res fail
        echo "Output differs for file $i. Diff follows:"
        diff -u tmp.out $i.out
    end

    if not diff tmp.err $i.err >/dev/null
        set res fail
        echo "Error output differs for file $i. Diff follows:"
        diff -u tmp.err $i.err
    end

    if test $tmp_status != (cat $i.status)
        set res fail
        echo "Exit status differs for file $i."
    end

    if test $res = ok
        echo "File $i tested ok"
    else
        echo "File $i failed tests"
    end
end
