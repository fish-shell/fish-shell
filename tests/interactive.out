File read.expect tested ok
